docarray/document/pydantic_model.py

```diff
@@ -54,6 +59,16 @@ class PydanticDocument(BaseModel):
     class Config:
         smart_union = True
 
+    def __init__(self, **data):
+        super().__init__(**data)
+        # underscore attributes need to be set and validated manually
+        _metadata = data.get('_metadata', None)
+        if _metadata is not None:
+            _MetadataModel(metadata=_metadata)  # validate _metadata
+            object.__setattr__(
```

Comment:
what is this `object`? Is `object` as the Python object base class? why not call `self`?

Reply:
yes, it's the Python base object; calling on `self` does not work, because that would invoke `BaseModel`s implementation of `__setattr__`, which fails (I assume for the same reasons that we can't pass underscore attributes in the normal way in the first place).
The pydantic creator also recommends this way of doing it [here](https://github.com/samuelcolvin/pydantic/issues/655#issuecomment-570310120).